dimod/constrained.py

```diff
@@ -657,10 +733,20 @@ def substitute_self_loops(self) -> Dict[Variable, Variable]:
         Acts on the objective and constraints in-place.
 
         Returns:
-            A mapping from the integer variable labels to their introduced
+            Mapping from the integer variable labels to their introduced
             counterparts. The constraint enforcing :math:`j == i` uses
             the same label.
 
+        Examples:
+            >>> from dimod import Integer, ConstrainedQuadraticModel
+            >>> i = Integer('i')
+            >>> cqm = ConstrainedQuadraticModel()
+            >>> cqm.add_constraint(i*i <=3, label='i squared')
+            'i squared'
+            >>> cqm.substitute_self_loops()                      # doctest: +IGNORE_RESULT
```

Comment:
it's not really important, but why is this `# doctest: +IGNORE_RESULT` so far away?

Reply:
Often the code snippet has a few of these and I like them aligned on the right. In this case there's just one so it sticks out.